### PR TITLE
 Partial Documentation of MarioDream, CHR Separation

### PR DESCRIPTION
**Pull request by pepperpow**

### CODE REVIEW

File: src/prg-c-d.asm [hunk 1]
```diff
@@ -7,39 +7,29 @@
...
-	.dw MarioDream_Bed
-	.dw MarioDream_Bubble
-	.dw MarioDream_DoNothing
-	.dw MarioDream_EraseBubble1
-	.dw MarioDream_EraseBubble2
-	.dw MarioDream_EraseBubble3
-	.dw MarioDream_EraseBubble4
-	.dw MarioDream_EraseBubble5
-	.dw MarioDream_Palettes
+.include "src/mario_dream/dream_ptr_table.asm"
```
Comment (kmck) [inline src/prg-c-d.asm]: maybe it's better to leave the pointers in this file and just separate out the ppu data?

File: src/prg-c-d.asm [hunk 2]
```diff
@@ -50,234 +40,37 @@ sub_BANKC_801C:
...
-	.db $78
-	.db $74
-	.db $70
-	.db $7C
-	.db $70
-	.db $74
-	.db $78
-
+; ---------------------------------------------------------------------------
+;; PlayerXHi ;; <Dream_SceneCounter> = PlayerXHi ;; How many times this part of the scene plays
```
Comment (kmck) [inline src/prg-c-d.asm]: you can add multiple labels in `ram.asm` and use those names in here instead, which might make the code read more clearly

File: src/mario_dream/castroll_data.asm [hunk 3]
```diff
@@ -0,0 +1,914 @@
...
+CastRoll_Porcupo:
+	.db $D0, $3E, $00, $30
+	.db $D0, $3E, $00, $38 ; 4
+	.db $D0, $3E, $00, $40 ; 8
+	.db $D0, $3E, $00, $48 ; $C
+	.db $F9, $3E, $00, $30 ; $10
+	.db $F9, $30, $00, $38 ; $14
+	.db $F9, $32, $00, $40 ; $18
+	.db $F9 ; $1C
+byte_BANKC_8D5F:
```
Comment (kmck) [inline src/mario_dream/castroll_data.asm]: this looks like an extraneous auto label. i think you can remove this and reformat the lines up until `CastRoll_Tweeter` into groups of 4 bytes

File: src/mario_dream/castroll_data.asm [hunk 4]
```diff
@@ -0,0 +1,914 @@
...
+	.db $A8
+	.db $24
+	.db $90
+	.db $30
+	.db $98
+	.db $3E
+	.db $A0
+	.db $42
+	.db $A8
+byte_BANKC_93A6:
```
Comment (kmck) [inline src/mario_dream/castroll_data.asm]: is this the "The End" text?

**Comment by Xkeeper0** [comment]
Sorry for the delay in getting back to you, but @tompep / @kmck -- any thoughts no this at this point? It's been a while (and now there are some conflicts), but it doesn't appear that the comments you left were responded to.

**Comment by tompep** [comment]
Oops!  I will continue working on this, not used to github responses what with the chatroom.  Conflicts make this a big no, so I can continue working on it.

**Comment by pepperpow** [comment]
i'm gonna close this and resubmit...

**Pull request closed without merging**